Sync 'Track' and 'WebVTT' IDL files with Web-Spec

Ahmad-S792 opened the pull request:
#### 08c40bd4fc5a5aef8bc5e2a3c8f9989007e575e1
<pre>
Sync &apos;Track&apos; and &apos;WebVTT&apos; IDL files with Web-Spec

<a href="https://bugs.webkit.org/show_bug.cgi?id=260734">https://bugs.webkit.org/show_bug.cgi?id=260734</a>

Reviewed by Eric Carlson.

This patch is to add &apos;interface&apos; web-spec links to relevant &apos;track&apos; and &apos;VTT&apos; files
and do minor sync-up (i.e., adding missing ?) and add &apos;FIXME&apos; where applicable for future work.

* Source/WebCore/html/track/AudioTrack.idl:
* Source/WebCore/html/track/AudioTrackList.idl:
* Source/WebCore/html/track/DataCue.idl:
* Source/WebCore/html/track/TextTrack.idl:
* Source/WebCore/html/track/TextTrackCue.idl:
* Source/WebCore/html/track/TextTrackCueList.idl:
* Source/WebCore/html/track/TextTrackList.idl:
* Source/WebCore/html/track/TrackEvent.idl:
* Source/WebCore/html/track/VideoTrack.idl:
* Source/WebCore/html/track/VideoTrackList.idl:
* Source/WebCore/html/track/VTTCue.idl:
* Source/WebCore/html/track/VTTRegion.idl:

Canonical link: <a href="https://commits.webkit.org/267323@main">https://commits.webkit.org/267323@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/855d6982f3159250949ddb36f86e5606be90d816

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/16289 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/16608 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/17027 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/18057 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/15276 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/16477 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/19688 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/16721 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/17655 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/16483 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/16921 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/13947 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/18824 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/14166 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/14745 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/21562 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/15153 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/14910 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/18222 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/15503 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/13136 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/14722 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3895 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/19088 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/15330 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

eric-carlson commented on Source/WebCore/html/track/TextTrackCue.idl:
```diff
@@ -33,17 +35,17 @@
     attribute EventHandler onenter;
     attribute EventHandler onexit;
 
+    [CallWith=CurrentDocument] constructor(double startTime, double endTime, DocumentFragment cueNode);
```
This should probably have a FIXME to make `endTime` unrestricted

Ahmad-S792 replied:
Will look again in web-spec, might be oversight on my end.

Ahmad-S792 replied:
Done. :-)

eric-carlson commented on Source/WebCore/html/track/AudioTrackList.idl:
```diff
@@ -31,8 +33,10 @@
 ] interface AudioTrackList : EventTarget {
     readonly attribute unsigned long length;
+
+    // FIXME: Remove 'item' from below to align with Web-Spec.
```
I think it would be really helpful if every FIXME had the url of a bugzilla

Ahmad-S792 replied:
Will try to raise 'FIXME' bug as well. :-)

cdumez replied:
Can you please replace `Web-Spec` with `the specification` in all your comments?

Ahmad-S792 replied:
Done.

eric-carlson reviewed (approved):
r=me once the bots are happy

cdumez commented on Source/WebCore/html/track/TextTrack.idl:
```diff
@@ -51,6 +54,7 @@ enum TextTrackKind { "subtitles", "captions", "descriptions", "chapters", "metad
 
     attribute EventHandler oncuechange;
 
+    // FIXME: Remove 'non-standard'
```
This is actually from https://dvcs.w3.org/hg/text-tracks/raw-file/default/608toVTT/region.html#extension-of-the-texttrack-object AFAICT?

Ahmad-S792 replied:
I am going by following: https://github.com/w3c/webvtt/pull/31 , which is to remove them from TextTrack.

Ahmad-S792 replied:
and https://github.com/w3c/webvtt/commit/970b3a2d425d26e63fe2cd94b030f8f80289af00

Ahmad-S792 committed:
Sync 'Track' and 'WebVTT' IDL files with Web-Spec

https://bugs.webkit.org/show_bug.cgi?id=260734

Reviewed by Eric Carlson.

This patch is to add 'interface' web-spec links to relevant 'track' and 'VTT' files
and do minor sync-up (i.e., adding missing ?) and add 'FIXME' where applicable for future work.

* Source/WebCore/html/track/AudioTrack.idl:
* Source/WebCore/html/track/AudioTrackList.idl:
* Source/WebCore/html/track/DataCue.idl:
* Source/WebCore/html/track/TextTrack.idl:
* Source/WebCore/html/track/TextTrackCue.idl:
* Source/WebCore/html/track/TextTrackCueList.idl:
* Source/WebCore/html/track/TextTrackList.idl:
* Source/WebCore/html/track/TrackEvent.idl:
* Source/WebCore/html/track/VideoTrack.idl:
* Source/WebCore/html/track/VideoTrackList.idl:
* Source/WebCore/html/track/VTTCue.idl:
* Source/WebCore/html/track/VTTRegion.idl:

Canonical link: https://commits.webkit.org/267323@main
